src/main/java/racingcar/domain/Cars.java

```diff
@@ -0,0 +1,79 @@
+package racingcar.domain;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import racingcar.domain.strategy.MoveStrategy;
+
+public class Cars {
+
+    private final List<Car> cars;
+
+    public Cars(List<Car> cars) {
+        verifyMinSize(cars);
+        verifyDuplicate(cars);
+        this.cars = new ArrayList<>(cars);
+    }
+
+    public static Cars of(List<String> nameList) {
+        List<Car> cars = new ArrayList<>();
+        for (String name : nameList) {
+            cars.add(new RacingCar(name));
+        }
+        return new Cars(cars);
+    }
+
+    private void verifyMinSize(List<Car> cars) {
+        if (cars.isEmpty()) {
+            throw new IllegalStateException();
+        }
+    }
+
+    private void verifyDuplicate(List<Car> cars) {
+        Set<String> carNames = cars.stream()
+            .map(Car::getName)
+            .collect(Collectors.toSet());
```

Comment:
아래와 같이 변경해볼 수 있을 것 같아요. 🤗
```suggestion
        Set<Car> carNames = new HashSet<>(cars);
```

제가 제안한대로 변경했는데도 예외 처리 로직이 제대로 동작하지 않을거에요. 👀 
혹시 무엇 때문인지 짐작이 가시나요? 🤗 

Set과 List 자료형에 대한 특징들을 공부해보시면 좋을 것 같아요. 😃